### ISSUE TITLE
Support for non-numeric values in md.pattern

### ISSUE THREAD

**Issue by lorenzwalthert** [stefvanbuuren/mice#83]
Dear @stefvanbuuren, thanks for maintaining and developing this package. When a dataset does not contain missing values, it seems that the first column is missing.
``` r
library(mice)
#> Lade nötiges Paket: lattice
md.pattern(mtcars)
#>      mpg cyl disp hp drat wt qsec vs am gear carb  
#> [1,]   1   1    1  1    1  1    1  1  1    1    1 0
#> [2,]   0   0    0  0    0  0    0  0  0    0    0 0
```

However, my primary interest was to use `md.pattern()` to analyze missing patterns in a data frame with both numeric and non-numeric columns, which does not give the expected result. 
```r
mtcars$disp <- rep(letters, 2)[1:32]
mtcars$disp[c(1, 5, 23)] <- NA
md.pattern(mtcars)
#> Warning in data.matrix(x): NAs durch Umwandlung erzeugt
#>      mpg cyl hp drat wt qsec vs am gear carb disp   
#> [1,]   1   1  1    1  1    1  1  1    1    1    0  1
#> [2,]   0   0  0    0  0    0  0  0    0    0   32 32
```
I see you are using `model.matrix(x)` in the source code pretty high-up in the call, which I think is where the unexpected behaviour is triggered, but I don't see why `md.pattern()` should be restricted to numerical data. Thanks.

**Comment by stefvanbuuren** [comment]
Thanks. I assume you are using the CRAN version `mice 2.46.0`. 

We've recently dumped `data.matrix()` from `md.pattern()`. The version in `2.47.1` also does not work on complete data: 
```
md.pattern(mtcars)
Error in `rownames<-`(`*tmp*`, value = table(pat)) : 
  attempt to set 'rownames' on an object with no dimensions
```
This is a bug we need to resolve.

**Comment by lorenzwalthert** [comment]
Yes, I use the CRAN version 2.46. Thanks for the quick response. The second part of my example indeed works with the devel version 2.47. Sorry I did not check -.- I renamed the issue to this effect.
